packages/verifiable-credentials/src/credentials.ts

```diff
@@ -128,29 +144,50 @@ export default class Credentials {
 	 * @param data 
 	 * @returns 
 	 */
-	async createCredentialJWT(data: any): Promise<object> {
+	async createCredentialJWT(subjectId: string, data: any, expirationDate?: string): Promise<object> {
+		// Ensure a credential schema has been specified
+		if (!data.schema) {
+			throw new Error('No schema specified')
+		}
+
+		// Ensure data matches specified schema
+		const schema = await this.context.getClient().getSchema(data.schema)
+		const isValid = await schema.validate(data);
+
+		// @todo: Check the schema is a "credential" schema type?
+
+		if (!isValid) {
+			throw new Error('Data does not match specified schema')
+		}
+
 		const issuer = await this.createIssuer();
 		const account = this.context.getAccount();
 		const did = await account.did();
 
-		const credential = {
+		const vcPayload: any = {
 			'@context': [
 				'https://www.w3.org/2018/credentials/v1',
 				'https://www.w3.org/2018/credentials/examples/v1',
 			],
-			id: '',
+			sub: subjectId,
 			type: ['VerifiableCredential'],
 			issuer: did,
 			issuanceDate: new Date().toISOString(),
```

Comment:
Is this correct? 

In some cases the `issuanceDate` could be set externally - eg it is valid from the date payment is received, which could be prior to the VC actually being issued here. This could be important in cases where things like fines are being issued based on the date the credential is paid for.

Reply:
Agreed. Should support `issuanceDate` option.

@cmcWebCode40 Perhaps you refactor to support an `options` parameter that replaces `expirationDate`. There can be two options; `expirationDate`, `issuanceDate`. If `issuanceDate` is not defined, default to now.

Reply:
I have added the `options` parameter and they are optional

Reply:
This looks good, can you also add a test to confirm the issuance date in the generated VC is the one sent in the options.

Reply:
@tahpot, I have added this test.